README.md

```diff
@@ -8,44 +8,47 @@
 
 **Table of Contents**
 
-- [CAPE](#cape)
+- [Configurable Asset Privacy for Ethereum (CAPE)](#configurable-asset-privacy-for-ethereum-cape)
   - [Obtaining the source code](#obtaining-the-source-code)
   - [Providing feedback](#providing-feedback)
+- [Documentation](#documentation)
+  - [CAP protocol specification](#cap-protocol-specification)
+  - [CAPE Contract specification](#cape-contract-specification)
 - [Environment](#environment)
   - [1. Install nix](#1-install-nix)
   - [2. Activate the nix environment](#2-activate-the-nix-environment)
   - [3. Verify installation](#3-verify-installation)
   - [4. direnv (Optional, but recommended for development)](#4-direnv-optional-but-recommended-for-development)
 - [Build](#build)
-- [Develop](#develop)
-- [Test](#test)
+- [Development](#development)
+  - [Running Tests](#running-tests)
+  - [Interactive development](#interactive-development)
+  - [Linting & Formatting](#linting--formatting)
+  - [Updating dependencies](#updating-dependencies)
+  - [Rust specific development notes](#rust-specific-development-notes)
+  - [Ethereum contracts](#ethereum-contracts)
   - [Testing against go-ethereum node](#testing-against-go-ethereum-node)
   - [Testing against hardhat node](#testing-against-hardhat-node)
     - [Separate hardhat node](#separate-hardhat-node)
     - [Hardhat node integrated in test command](#hardhat-node-integrated-in-test-command)
-  - [Deployment](#deployment)
-    - [Linking to deployed contracts](#linking-to-deployed-contracts)
-  - [Precompiled solidity binaries](#precompiled-solidity-binaries)
-    - [Details about solidity compiler (solc) management](#details-about-solidity-compiler-solc-management)
-  - [Ethereum contracts](#ethereum-contracts)
-  - [Rust](#rust)
-  - [Linting & Formatting](#linting--formatting)
-  - [Updating dependencies](#updating-dependencies)
-  - [Alternative nix installation methods](#alternative-nix-installation-methods)
-    - [Nix on debian/ubuntu](#nix-on-debianubuntu)
-  - [Git hooks](#git-hooks)
-  - [Ethereum key management](#ethereum-key-management)
-  - [Python tools](#python-tools)
+- [Deployment](#deployment)
+  - [Linking to deployed contracts](#linking-to-deployed-contracts)
+- [Precompiled solidity binaries](#precompiled-solidity-binaries)
+  - [Details about solidity compiler (solc) management](#details-about-solidity-compiler-solc-management)
+- [Alternative nix installation methods](#alternative-nix-installation-methods)
+  - [Nix on debian/ubuntu](#nix-on-debianubuntu)
+    - [Installation](#installation)
+    - [Uninstallation](#uninstallation)
+- [Git hooks](#git-hooks)
+- [Ethereum key management](#ethereum-key-management)
+- [Python tools](#python-tools)
 - [Rinkeby](#rinkeby)
 - [Goerli](#goerli)
 - [Gas Usage](#gas-usage)
   - [Gas Reporter](#gas-reporter)
   - [Gas usage of block submissions](#gas-usage-of-block-submissions)
 - [CI](#ci)
   - [Nightly CI builds](#nightly-ci-builds)
```

Comment:
maybe: 
- all of these can be moved under the umbrella of `# Development` 
- merge `Rinkeby` and `Goerli` under `Deployments` as `testnet specific detail`

Reply:
Thanks. I tried to group it as suggested.